projects/xrsh/programs/xrsh/tests/basic.nix

```diff
@@ -0,0 +1,32 @@
+{
+  sources,
+  ...
+}:
+
+{
+  name = "xrsh";
+
+  nodes = {
+    machine =
+      { ... }:
+      {
+        imports = [
+          sources.modules.ngipkgs
+          sources.modules.programs.xrsh
+          sources.examples.xrsh.basic
+        ];
+      };
+  };
+
+  testScript =
+    { nodes, ... }:
+    ''
+      start_all()
+
+      machine.succeed("xrsh >&2 &")
+
+      # xrsh serves defaultly on :8080
+      machine.wait_for_open_port(8090)
+      machine.succeed("curl -i 0.0.0.0:8090")
```

Comment:
Does the machine shut down after the command ends and the service is destroyed? Or is there a potential this will keep the VM around until some horribly long timeout?

Reply:
We tested it yesterday and I believe that it did shut down.